### PR TITLE
improvement: Merge input and input3 projects used for tests

### PR DESCRIPTION
**Pull request by tgodzik**
Previously, we would use input with Scala 2 compatible files in Scala 2 specific test and input3 with different files for Scala 3 tests.

Now, we merged all the sources with the incompatible ones in scala-2 or scala-3 folders. We still have two projects, but we will test more files from both Scala 3 and Scala 2 perspective.

### CODE REVIEW

File: tests/unit/src/test/resources/definition-scala3/example/Scalalib.scala [hunk 1]
```diff
@@ -0,0 +1,26 @@
+package example
+
+class Scalalib/*Scalalib.scala*/ {
+  val nil/*Scalalib.scala*/ = List/*package.scala*/()
+  val lst/*Scalalib.scala*/ = List/*package.scala*/[
+    (
+        Nothing,
```
Comment (tgodzik) [inline tests/unit/src/test/resources/definition-scala3/example/Scalalib.scala:7]: Very similar as for Scala 2, just for `==` we don't have def, but this doesn't seem ncessary.

File: tests/unit/src/test/resources/documentSymbol-scala3/example/Extension.scala [hunk 2]
```diff
@@ -1,15 +1,11 @@
-/*example(Package):15*/package example
+/*example(Package):11*/package example
 
-/*example.extension Int(Module):4*/extension (i: Int)
```
Comment (tgodzik) [inline tests/unit/src/test/resources/documentSymbol-scala3/example/Extension.scala:3]: This got reformatted by mistake, but looks correct.

File: tests/unit/src/test/resources/definition-scala3/example/ForComprehensions.scala [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    ) == (1, 2)
+    (
+      c/*ForComprehensions.semanticdb*/,
+      d/*ForComprehensions.semanticdb*/,
+    ) <- List/*package.scala*/((a/*ForComprehensions.semanticdb*/, b/*ForComprehensions.semanticdb*/))
+    if (
+      a/*ForComprehensions.semanticdb*/,
+      b/*ForComprehensions.semanticdb*/,
+      c/*ForComprehensions.semanticdb*/,
+      d/*ForComprehensions.semanticdb*/,/*unexpected: local4*/
```
Comment (tgodzik) [inline tests/unit/src/test/resources/definition-scala3/example/ForComprehensions.scala:19]: `unexpected` comes from https://github.com/lampepfl/dotty/issues/16872

File: tests/unit/src/test/resources/definition-scala3/example/NamedArguments.scala [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+  // anonymous classes
+  @deprecated/*deprecated.scala*/(
+    message/*deprecated.scala fallback to scala.deprecated#*/ = "a",
+    since/*deprecated.scala fallback to scala.deprecated#*/ = susan/*NamedArguments.scala*/,/*unexpected: example.NamedArguments.susan.*/
+  ) def b/*NamedArguments.scala*/ = 1
+
+  // vararg
+  List/*package.scala*/(
+    elems/*<no symbol>*/ = 2
```
Comment (tgodzik) [inline tests/unit/src/test/resources/definition-scala3/example/NamedArguments.scala:31]: https://github.com/lampepfl/dotty/issues/16911

File: tests/unit/src/test/resources/definition-scala3/example/StructuralTypes.scala [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+
+  val user/*StructuralTypes.scala*/ = null.asInstanceOf[User/*StructuralTypes.scala*/]
+  user/*StructuralTypes.scala*/.name/*<no symbol>*/
+  user/*StructuralTypes.scala*/.age/*<no symbol>*/
+
+  val V/*StructuralTypes.scala*/: Object/*Object.java*/ {
+    def scalameta/*StructuralTypes.semanticdb*/: String/*Predef.scala*/
+  } = new:
+    def scalameta/*StructuralTypes.semanticdb*/ = "4.0"
+  V/*StructuralTypes.scala*/.scalameta/*<no symbol>*/
```
Comment (tgodzik) [inline tests/unit/src/test/resources/definition-scala3/example/StructuralTypes.scala:19]: https://github.com/lampepfl/dotty/issues/16912

**Comment by tgodzik** [comment]
Note for reviewers:

The first commit adds input3 as a separate cross compiled project, which is why we get a lot of new expect cases. I recommend looking into the sbt file only to see the changes and then looking at the other commits. I went through all the cases to check if there is anything unexpected and fixed it in those successive commits

File: build.sbt [hunk 6]
```diff
@@ -516,6 +516,13 @@ lazy val input3 = project
   .settings(
     sharedSettings,
     scalaVersion := V.scala3,
+    target := (ThisBuild / baseDirectory).value / "input" / "target" / "target3",
+    Compile / unmanagedSourceDirectories := Seq(
+      (ThisBuild / baseDirectory).value / "tests" / "input" / "src" / "main" / "scala",
+      (ThisBuild / baseDirectory).value / "tests" / "input" / "src" / "main" / "scala-3",
+      (ThisBuild / baseDirectory).value / "tests" / "input" / "src" / "main" / "java",
```
Comment (dos65) [inline build.sbt:523]:
```suggestion
      (input / baseDirectory).value / "src" / "main" / "scala",
      (input / baseDirectory).value / "src" / "main" / "scala-3",
      (input / baseDirectory).value /  "src" / "main" / "java",
```

File: .scalafmt.conf [hunk 7]
```diff
@@ -20,6 +20,7 @@ project.excludeFilters = [
   "metals-bench/src/main/resources"
   "tests/unit/src/test/resources"
   "tests/input3/src/main/scala"
+  "tests/input/src/main/scala"
```
Comment (dos65) [inline .scalafmt.conf:23]: The line about with `input3` might be removed

**Review by dos65 — APPROVED** [review]
LGTM. Left just few suggestions

File: tests/unit/src/test/scala/tests/DefinitionSuite.scala [hunk 8]
```diff
@@ -48,6 +48,7 @@ abstract class DefinitionSuiteBase(
         ScalaVersions.dialectForDependencyJar(jar.filename),
       )
     }
+    List(elems = 1)
```
Comment (kasiaMarek) [inline tests/unit/src/test/scala/tests/DefinitionSuite.scala:51]: Why is this here?

File: mtags/src/main/scala/scala/meta/internal/mtags/DefinitionAlternatives.scala [hunk 9]
```diff
@@ -33,6 +34,19 @@ object DefinitionAlternatives {
       GlobalSymbol(owner, Descriptor.Type(name))
     }
 
+  private val objectMethods = Set("equals", "toString", "getClass", "hashCode")
+
+  /**
+   * If Any is used we don't have any source to go to, so instead we try with java Object.
+   */
+  private def objectInsteadOfAny(symbol: Symbol): Option[Symbol] = {
```
Comment (kasiaMarek) [inline mtags/src/main/scala/scala/meta/internal/mtags/DefinitionAlternatives.scala:42]: It seems a bit misleading, though I guess it is reasonable enough.

File: tests/unit/src/test/scala/tests/DefinitionSuite.scala (hunk 8 above)
Comment (tgodzik) [reply tests/unit/src/test/scala/tests/DefinitionSuite.scala:51]: BecauseI was experimenting and totally forgot :sweat_smile: Thanks for noticing!

Comment (tgodzik) [reply tests/unit/src/test/scala/tests/DefinitionSuite.scala:51]: Fixed

File: .scalafmt.conf (hunk 7 above)
Comment (tgodzik) [reply .scalafmt.conf:22]: Done, thanks!

File: build.sbt (hunk 6 above)
Comment (tgodzik) [reply build.sbt]: Good idea, changed.

File: mtags/src/main/scala/scala/meta/internal/mtags/DefinitionAlternatives.scala (hunk 9 above)
Comment (tgodzik) [reply mtags/src/main/scala/scala/meta/internal/mtags/DefinitionAlternatives.scala:42]: Any ideas for an alternative name? I can't come up with anything clearer :sweat:
